stage0_dice/src/lib.rs

```diff
@@ -168,6 +171,18 @@ pub fn generate_dice_data<
     let report = get_attestation(report_data).expect("couldn't get attestation report.");
     let report_bytes = report.as_bytes();
 
+    // Use the root derived key as the UDS (unique device secret) for deriving sealing keys.
+    let ikm: DerivedKey = get_derived_key().expect("couldn't get derived key");
+
+    // Mix in the measurements of the kernel, the kernel command-line, the kernel setup data and the
+    // initial RAM disk when deriving the CDI for Layer 1.
+    let mut salt: Vec<u8> = Vec::with_capacity(128);
+    salt.extend_from_slice(&measurements.kernel_measurement[..]);
+    salt.extend_from_slice(&measurements.cmdline_measurement[..]);
+    salt.extend_from_slice(&measurements.setup_data_measurement[..]);
+    salt.extend_from_slice(&measurements.ram_disk_measurement[..]);
+    let hkdf = Hkdf::<Sha256>::new(Some(&salt), &ikm[..]);
```

Comment:
Is this additional entropy for the next layer key necessary? And does this key need to be deterministic? I thought it could just be randomly generated. Is it because for sealing we expect the key to be deterministic?

Reply:
Yes, it must be deterministic to support sealing and unsealing across reboots. If any of these measurements change, it could indicate a different behaviour of the workload, so we want a different key to be generated.